### PR TITLE
langchain[patch]: structured output chain nits

### PR DESCRIPTION
**Pull request by baskaryan**

### CODE REVIEW

**Review by hinthornw — APPROVED** [review]
I like!

File: libs/langchain/langchain/chains/structured_output/base.py [hunk 1]
```diff
@@ -103,17 +95,22 @@ class RecordDog(BaseModel):
...
     output_schema: Union[Dict[str, Any], Type[BaseModel]],
     llm: Runnable,
-    prompt: BasePromptTemplate,
+    prompt: Optional[BasePromptTemplate] = None,
     *,
     output_parser: Optional[Union[BaseOutputParser, BaseGenerationOutputParser]] = None,
-    mode: Literal["openai-functions", "openai-json"] = "openai-functions",
+    mode: Literal[
+        "openai-tools", "openai-functions", "openai-json"
+    ] = "openai-functions",
```
Comment (eyurtsev) [inline libs/langchain/langchain/chains/structured_output/base.py:113]: I wonder whether we should've made this a required argument

**Review by eyurtsev — APPROVED** [review]
These changes look good to me. I think discovery of this chain is hard because the more intuitive names of extraction chain are already taken. Maybe we could add a deprecation decorator after i update the use case docs?

File: libs/langchain/langchain/chains/structured_output/base.py (hunk 1 above)
Comment (baskaryan) [reply libs/langchain/langchain/chains/structured_output/base.py:113]: yea, couldn't bc of backwards compat (mode arg was added after the function was created)

File: libs/langchain/langchain/chains/structured_output/base.py [hunk 2]
```diff
@@ -103,17 +95,22 @@ class RecordDog(BaseModel):
...
 def create_structured_output_runnable(
     output_schema: Union[Dict[str, Any], Type[BaseModel]],
     llm: Runnable,
-    prompt: BasePromptTemplate,
+    prompt: Optional[BasePromptTemplate] = None,
     *,
     output_parser: Optional[Union[BaseOutputParser, BaseGenerationOutputParser]] = None,
-    mode: Literal["openai-functions", "openai-json"] = "openai-functions",
+    mode: Literal[
+        "openai-tools", "openai-functions", "openai-json"
```
Comment (baskaryan) [inline libs/langchain/langchain/chains/structured_output/base.py:112]: should delete openai-tools, not part of this pr
